src/payment/strategies/afterpay/afterpay-payment-strategy.ts

```diff
@@ -49,16 +50,20 @@ export default class AfterpayPaymentStrategy implements PaymentStrategy {
     }
 
     async execute(payload: OrderRequestBody, options?: PaymentRequestOptions): Promise<InternalCheckoutSelectors> {
-        const paymentId = payload.payment && payload.payment.gatewayId;
+        if (!payload.payment) {
+            throw new PaymentArgumentInvalidError(['payment.gatewayId', 'payment.methodId']);
+        }
+        const { gatewayId, methodId } = payload.payment;
 
-        if (!paymentId) {
-            throw new PaymentArgumentInvalidError(['payment.gatewayId']);
+        if (!gatewayId || !methodId) {
+            throw new PaymentArgumentInvalidError(['payment.gatewayId', 'payment.methodId']);
         }
 
         let state = this._store.getState();
         const currencyCode = state.cart.getCart()?.currency.code || '';
         const countryCode = this._mapCurrencyToISO2(currencyCode);
-        const { useStoreCredit } = payload;
+        let { useStoreCredit } = payload;
+        useStoreCredit = typeof useStoreCredit === 'undefined' ? this._store.getState().checkout.getCheckoutOrThrow().isStoreCreditApplied : useStoreCredit;
```

Comment:
I think this condition will cause `applyStoreCredit` call below to fire, even if there's no need to. You only need to call `applyStoreCredit` if `payload.useStoreCredit` is defined. What you have now is, if `payload.useStoreCredit` is undefined, call `applyStoreCredit` again with the same value that is currently set for the checkout session, which is redundant. As far as I can tell from reading the code, I don't think you need to make a change at all to achieve the goal of this PR?

Reply:
indeed, this change comes from this PR https://github.com/bigcommerce/checkout-sdk-js/pull/1209 which is needed because some of the changes that they are introducing there